mathesar/rpc/data_modeling.py

```diff
@@ -104,3 +104,35 @@ def suggest_types(*, table_oid: int, database_id: int, **kwargs) -> dict:
     user = kwargs.get(REQUEST_KEY).user
     with connect(database_id, user) as conn:
         return infer_types.infer_table_column_data_types(conn, table_oid)
+
+
+@rpc_method(name="data_modeling.split_table")
+@http_basic_auth_login_required
+@handle_rpc_exceptions
+def split_table(
+    *,
+    table_oid: int,
+    column_attnums: list,
+    extracted_table_name: str,
+    relationship_fk_column_name: str,
+    database_id: int,
+    **kwargs
+) -> None:
+    """
+    Extract columns from a table to create a new table, linked by a foreign key.
+
+    Args:
+        table_oid: The OID of the table whose columns we'll extract.
+        column_attnums: A list of the attnums of the columns to extract.
+        extracted_table_name: The name of the new table to be made from the extracted columns.
+        relationship_fk_column_name: The name to give the new foreign key column in the remainder table (optional)
```

Comment:
It's not an optional param the way this function is defined.

As for whether it _should_ be, how is it now in the UI?